core/src/main/java/org/jobrunr/utils/JobUtils.java

```diff
@@ -60,7 +60,14 @@ public static Method getJobMethod(Class<?> jobClass, JobDetails jobDetails) {
     }
 
     public static void assertJobExists(JobDetails jobDetails) {
-        assertJobExists(getJobSignature(jobDetails));
+        String className = getJobClassName(jobDetails);
+        String methodName = jobDetails.getMethodName();
+        String[] jobParameterTypesNames = getDeserializableJobParametersForSignature(jobDetails);
+        if (jobParameterTypesNames.length == jobDetails.getJobParameters().size()) {
+            assertJobExists(className, methodName, jobParameterTypesNames);
+        } else {
+            throw new JobMethodNotFoundException(jobDetails);
+        }
     }
 
     public static void assertJobExists(String jobSignature) {
```

Comment:
This method (`assertJobExists(String jobSignature)`) was added recently and is no longer used internally other than in tests other than in tests. Should we remove it?